### ISSUE TITLE
Unable to open camera

### ISSUE THREAD

**Issue by JaimeFerBec** [code-kotis/barcode-scanner#7]
Whenever I open the app, a toast message shows at the bottom that says: "Unable to open camera, provide permission to access the camera".
The pwa successfully accessed the camera before, but somehow its permissions got revoked.
How do you give permissions to a particular progressive web app?

**Comment by gokulkrishh** [comment]
@JaimeFerBec Can you pls post the which device and the browser you are using??

And for permission for camera is invoked via https://github.com/code-kotis/barcode-scanner/blob/master/app/js/vendor/qrscan.js#L44

**Comment by JaimeFerBec** [comment]
**Device**: Google Nexus 4 Build/LMY48T
**OS**: Android 5.1.1
**Web browser**: Google Chrome 54.0.2840.85

**Comment by JaimeFerBec** [comment]
This is weird: I just accessed the site's configuration on Chrome by clicking on the (i)nfo icon that appears if you first click the vertical three dots icon in the top-right corner of the browser, and it says the site has permission to access the camera.

**EDIT**: I deleted all the data the browser had stored for qrcodescan.in, including the permission to use the camera. I did so to test if it would ask me for permission to use my camera again after reloading the webpage, but it didn't. It showed the same toast message that I mentioned on my first comment.

**Comment by gokulkrishh** [comment]
@JaimeFerBec Weird issue. It is supposed to work, might be due to nexus 4 device but not sure. Since you are using latest chrome browser. I will take a look into it. Thanks for reporting 👍

**Comment by JaimeFerBec** [comment]
I opened the progressive web app again this morning, and it prompted me for camera permission again. After I gave permission for it to use the camera, I have used it several times during this morning, an the failure has not come up again.
The reason why it asked me for permission again remains unknown, though. Yesterday I tried cleaning Chrome's cache and also deleting all the data related to qrscan.in the browser had stored. The only different thing I did was turning the phone off last night. I guess that cleaned something up, making it possible por the PWA to ask for camera permission again.
